src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -0,0 +1,154 @@
+                appAddress,
+                SecretOwnerRole.APPLICATION_DEVELOPER,
+                null,
+                secretIndex,
+                secretValue
+        );
+        return ResponseEntity.noContent().build();
+    }
+
+    @RequestMapping(method = RequestMethod.HEAD, path = "/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<Void> isAppRuntimeSecretPresent(@PathVariable String appAddress,
+                                                          @PathVariable long secretIndex) {
+        final boolean isSecretPresent = teeTaskRuntimeSecretService.isSecretPresent(
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.APPLICATION_DEVELOPER,
+                null,
+                secretIndex
+        );
+        if (isSecretPresent) {
+            log.info("App developer secret found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+            return ResponseEntity.noContent().build();
+        }
+
+        log.info("App developer secret not found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+        return ResponseEntity.notFound().build();
+    }
+
+    @PostMapping("/{appAddress}/requesters/secrets")
+    public ResponseEntity<Map<String, String>> setRequesterSecretCountForApp(
+            @RequestHeader("Authorization") String authorization,
+            @PathVariable String appAddress,
+            @RequestBody int secretCount) {
+        String challenge = authorizationService
+                .getChallengeForSetAppRequesterRuntimeSecretCount(
+                        appAddress,
+                        secretCount
+                );
+
+        if (!authorizationService.isSignedByOwner(challenge, authorization, appAddress)) {
+            log.error("Unauthorized to setRequesterSecretCountForApp" +
+                            " [appAddress: {}, expectedChallenge: {}]",
+                    appAddress, challenge);
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+
+        final boolean isCountAlreadyPresent = teeTaskRuntimeSecretCountService
+                .isAppRuntimeSecretCountPresent(
+                        appAddress,
+                        SecretOwnerRole.REQUESTER
+                );
+        if (isCountAlreadyPresent) {
+            log.info("Can't add app requester app secret count as it already exist"
+            + " [appAddress:{}]", appAddress);
+            return ResponseEntity.status(HttpStatus.CONFLICT).build(); // secret count already exists
+        }
+
+        final boolean hasBeenInserted = teeTaskRuntimeSecretCountService.setAppRuntimeSecretCount(
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                secretCount
+        );
+
+        if (!hasBeenInserted) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(createErrorPayload(
```

Comment:
Up @mcornaton 
Do we need error metadata since each error has its own code?

Reply:
If you take a look at `addRequesterAppRuntimeSecret`, you'll see that `BAD_REQUEST` can be returned in a bunch of places:
- when `secretIndex > 0` - this will probably be removed, or updated to 9 if we want to keep a max limit ;
- when `secretIndex < 0` ;
- when no max secret count has been provided ;
- when index is greater than max secret count.
There, we need metadata. In order to be consistent, it seems great to add the error description to the return body. [It would also allow to extend the API later, as mentioned by @zguesmi](https://github.com/iExecBlockchainComputing/iexec-sms/pull/69#discussion_r758217950):
> If we want to send error messages in the response body, let's use a common JSON format for all endpoints. It makes it predictable for clients and extendable in the future without breaking the API.

Reply:
I agree with @jeremyjams for the `500 INTERNAL_SERVER_ERROR`. It makes more sense.

Reply:
I disagree on the `500 INTERNAL_SERVER_ERROR`. It's triggered when the `secretIndex` is negative, which is a user mistake. It's nowhere due to the server.
BTW, it should probably be checked in the controller before the call to the service.

Reply:
Yes it should be checked in the controller beforehand.

Reply:
Validation done in https://github.com/iExecBlockchainComputing/iexec-sms/pull/69/commits/db53dc3b8ee44040374a1e970701c23f778dbab5 :)

Reply:
Thanks!